code/game/machinery/engi_adv.dm

```diff
@@ -0,0 +1,245 @@
+// Construction "nuke"
+
+/obj/machinery/construction_nuke
+	name = "nuclear fission construction device"
+	desc = "The next level of interior redecoration."
+	icon = 'icons/obj/machines/nuke.dmi'
+	icon_state = "nuclearbomb0"
+	density = TRUE
+	use_power = 0
+
+	var/timer_set = 90
+	var/ui_style = "nanotrasen"
+	var/range = 160
+
+	var/timing = FALSE
+	var/detonation_timer
+	var/cooldown = 0
+	var/safety = TRUE
+
+	var/bomb_set = FALSE
+	var/exploding = FALSE
+	var/quiet = FALSE
+	var/payload = "plasteel"
+	var/payload_wall = /turf/closed/wall/r_wall
+	var/payload_floor = /turf/open/floor/engine
+	var/static/list/possible_payloads = list("wood","sand","ice","mining","silver","gold","bananium","abductor","desolation", "plasma","uranium","bluespace","diamond","plasteel","safety","titanium","plastitanium")
+
+/obj/machinery/construction_nuke/Initialize()
+	. = ..()
+	GLOB.poi_list += src
+
+/obj/machinery/construction_nuke/Destroy()
+	GLOB.poi_list -= src
+	return ..()
+
+/obj/machinery/construction_nuke/examine(mob/user)
+	. = ..()
+	if(timing)
+		to_chat(user, "There are [get_time_left()] seconds until detonation.")
+
+/obj/machinery/construction_nuke/update_icon()
+	. = ..()
+	if(safety)
+		icon_state = "nuclearbomb1"
+	else
+		icon_state = "nuclearbomb2"
+	if(timing)
+		if(get_time_left() <= 14)
+			icon_state = "nuclearbomb3"
+		else
+			icon_state = "nuclearbombc"
+	
+		
+
+/obj/machinery/construction_nuke/process()
+	if(timing)
+		bomb_set = TRUE
+		if(detonation_timer < world.time && !exploding)
+			explode()
+		else
+			switch(get_time_left())
+				if (30 to 3600)
+					playsound(src, 'sound/items/timer.ogg', 5, 0)
+				if (15 to 29)
+					playsound(src, 'sound/items/timer.ogg', 30, 0)
+				if (0 to 14)
+					update_icon()
+					quiet  = !quiet
+					if(!quiet)
+						playsound(src, 'sound/machines/engine_alert2.ogg', 100, 0)
+
+/obj/machinery/construction_nuke/interact(mob/user)
+	user.set_machine(src)
+	var/list/dat = list()
+	dat +="<div class='statusDisplay'>"
+	dat += "Timer: [get_time_left()] seconds<br>"
+	dat += "</div>"
+	dat += "<b><u>Detonation Payload</u>: <A href='?src=\ref[src];action=payload'>[payload]</A></b><br><br>"
+	dat += "<A href='?src=\ref[src];action=set'>Set Timer</A><br>"
+	dat += "<A href='?src=\ref[src];action=anchor'>[anchored ? "Anchored" : "Not Anchored"]</A><br>"
+	dat += "<A href='?src=\ref[src];action=safety'>[safety ? "Safety On" : "Safety Off"]</A><br><br>"
+	dat += "<b><A href='?src=\ref[src];action=activate'>[bomb_set ? "DEACTIVATE" : "ACTIVATE"]</A></b><br>"
+	var/datum/browser/popup = new(user, "vending", "Construction Nuke", 300, 275)
+	popup.set_content(dat.Join())
+	popup.open()
+
+
+/obj/machinery/construction_nuke/Topic(href, href_list)
```

Comment:
I'd prefer you use a tgui if possible so these can't be spoofed or at least add the sanity checks

Reply:
What sanity checks do I need?

Reply:
Adjacent, incapacitated, canUseTopic etc

Reply:
Done